src/transcriber/main.py

```diff
@@ -0,0 +1,133 @@
+# /// script
+# dependencies = [
+#   "click",
+#   "faster-whisper",
+#   "jinja2",
+#   "tqdm",
+# ]
+# ///
+
+import os
+import time
+import click
+import hashlib
+from datetime import datetime
+from pathlib import Path
+from tqdm import tqdm
+import sys
+
+# --- START_IMPORTS ---
+# Add current directory to sys.path to ensure local modules can be imported
+# even when running as a script or within Bazel structure.
+sys.path.append(os.path.dirname(os.path.abspath(__file__)))
+
+try:
+    from schemas import ModelInfo, FileInfo, TranscriptionMetadata, TranscriptionResult
+    from render import render_template
+except ImportError as e1:
+    try:
+        from src.transcriber.schemas import ModelInfo, FileInfo, TranscriptionMetadata, TranscriptionResult
+        from src.transcriber.render import render_template
+    except ImportError as e2:
+        # Check if classes are already defined (e.g. by inlining/concatenation)
+        if 'FileInfo' in globals():
+            pass
+        else:
+            raise ImportError(f"Imports failed. CWD: {os.getcwd()}. sys.path: {sys.path}. E1: {e1}. E2: {e2}")
+# --- END_IMPORTS ---
+
+def get_file_info(file_path: str) -> 'FileInfo':
+    path = Path(file_path)
+    stat = path.stat()
+    return FileInfo(
+        path=str(path.absolute()),
+        name=path.name,
+        size_bytes=stat.st_size,
+        duration_seconds=None # Populated later if possible
+    )
+
+@click.command()
+@click.argument('audio_file', type=click.Path(exists=True))
+@click.option('--template', '-t', type=click.Path(exists=True), help='Path to Jinja2 template file.')
+@click.option('--output', '-o', type=click.Path(), help='Output file path.')
+@click.option('--model', '-m', default='base', help='Whisper model size (tiny, base, small, medium, large-v3).')
+@click.option('--device', default='auto', help='Device to use (cuda, cpu, auto).')
+@click.option('--compute-type', default='default', help='Compute type (float16, int8_float16, int8, etc).')
+def transcribe(audio_file, template, output, model, device, compute_type):
+    """
+    Transcribe AUDIO_FILE using faster-whisper and render the result using a Jinja2 template.
+    """
+    start_time = time.time()
```

Comment:
We should start from transcription, not from lazily downloading the model 

We can also measure the time it takes to load the model